### ISSUE TITLE
Incompatibility with StimulusReflex >3.4

### ISSUE THREAD

**Issue by Zajn** [podia/stimulus_reflex_testing#17]
We recently updated to StimulusReflex 3.5.0.pre9, [which introduced changes](https://github.com/stimulusreflex/stimulus_reflex/pull/561) to the process of verifying identical client/server version numbers.

The version number is now verified when the channel receives data, so it required to be part of the reflex data.

I'm not intimately familiar with SR or this library, but I can try and get a PR together that would address this.

**Comment by jasoncharnes** [comment]
Thanks for pointing out this issue!

> I'm not intimately familiar with SR or this library, but I can try and get a PR together that would address this.

That would wonderful. I'm glad to provide any help I can along the way.

**Comment by Zajn** [comment]
> That would be wonderful. I'm glad to provide any help I can along the way.

Great! I'll see what I can do.

One idea that I had, which might be terrible, was to just simply update `build_reflex` to add the version of SR to the data it uses to instantiate the reflex that's being tested. Since SR now just [adds the version data](https://github.com/leastbad/stimulus_reflex/blob/ec10b2ed66fa403c80988d39e29616b400614089/javascript/reflex_data.js#L131) as part of the `ReflexData` class on the client.

```diff
    def build_reflex(opts = {})
      channel = opts.fetch(:channel, TestChannel.new(opts.fetch(:connection, {})))
      element = opts.fetch(:element, StimulusReflex::Element.new)
+     version = StimulusReflex::VERSION   

      self.class.reflex_class.new(
-       channel, element: element, url: opts.fetch(:url, ""), method_name: method_name_from_opts(opts), params: opts.fetch(:params, {})
+       channel, element: element, url: opts.fetch(:url, ""), method_name: method_name_from_opts(opts), params: opts.fetch(:params, {}), version: version
      )
    end
```

The obvious issue there is that it just assumes that the version of the NPM package and the gem would be in sync. Since this library I think is more focused on just testing the expected outcome of a reflex vs. actually triggering one from SR, that _seemed_ like an okay assumption.

What do you think?

**Comment by jasoncharnes** [comment]
I think that would work! 👍

**Comment by phacks** [comment]
In case that helps others, we worked around this issue by disabling this check in test mode (available since version `3.4.0` of SR):

```ruby
StimulusReflex.configure do |config|
  # Enable/disable exiting / warning when the sanity checks fail options:
  # `:exit` or `:warn` or `:ignore`
  config.on_failed_sanity_checks = Rails.env.test? ? :ignore : :warn
 
 # ...
end
```
